tools/pio/copy_files.py

```diff
@@ -23,6 +23,9 @@ def get_max_bin_size(env_name, file_suffix):
         if "factory" in file_suffix:
             # Factory bin files include a part which is not overwritten via OTA
             max_bin_size = max_bin_size + 65536
+    if "_16M8M" in env_name or "_16M2M" in env_name:
```

Comment:
Maybe also check for "ESP32" here?

Reply:
NP, can change later today, this felt more generic 😄

Reply:
Yep, but you can't have 4M sketch for ESP8266, but you can have 16M of flash on ESP8266.